src/backend/distributed/executor/multi_utility.c

```diff
@@ -2921,6 +2934,126 @@ ErrorIfUnsupportedTruncateStmt(TruncateStmt *truncateStatement)
 }
 
 
+/*
+ * ProcessTruncateStatement handles distributed locking
+ * of truncated tables before standard utility takes over.
+ *
+ * This is only for distributed serialization of truncate commands.
+ * The function assumes that there is no foreign key relation between
+ * non-distributed and distributed relations.
+ */
+static void
+ProcessTruncateStatement(TruncateStmt *truncateStatement)
+{
+	List *distributedRelationList = NIL;
+
+	ListCell *relationCell = NULL;
```

Comment:
Shall we move the loop into a separate function like: `LockTableInMetadataWorkers()`. My reasoning is that I'll expand `ProcessTruncateStatement ()` for fixing #2382 such that I can add another function here:
```C
def ProcessTruncateStatement():
      LockTableInMetadataWorkers()
      SetSequentialModeIfNecessary()
      /* and others in the future */
```

Also, we can return asap in `LockTableInMetadataWorkers ()`  when `! ClusterHasKnownMetadataWorkers ()`

Reply:
I am not fond of this change but added it.